integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,353 @@
+        hybrid_retrieval.add_component("text_embedder", self.embedder)
+        hybrid_retrieval.add_component("embedding_retriever", embedding_retriever)
+        hybrid_retrieval.add_component("bm25_retriever", bm25_retriever)
+        hybrid_retrieval.add_component("document_joiner", document_joiner)
+
+        hybrid_retrieval.connect("text_embedder.embedding", "embedding_retriever.query_embedding")
+        hybrid_retrieval.connect("bm25_retriever", "document_joiner")
+        hybrid_retrieval.connect("embedding_retriever", "document_joiner")
+
+        # Define how pipeline inputs/outputs map to subcomponent inputs/outputs
+        self.input_mapping = {
+            # The pipeline input "query" feeds into each of the retrievers
+            "query": ["text_embedder.text", "bm25_retriever.query"],
+        }
+        self.output_mapping = {"document_joiner.documents": "documents"}
+
+        return hybrid_retrieval
+
+    def to_dict(self):
+        """
+        Serialize OpenSearchHybridRetriever to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            # DocumentStore
+            document_store=self.document_store.to_dict(),
+            embedder=self.embedder.to_dict(),
+            filters_bm25=self.filters_bm25,
+            fuzziness=self.fuzziness,
+            top_k_bm25=self.top_k_bm25,
+            scale_score=self.scale_score,
+            all_terms_must_match=self.all_terms_must_match,
+            filter_policy_bm25=self.filter_policy_bm25.value,
+            custom_query_bm25=self.custom_query_bm25,
+            # OpenSearchEmbeddingRetriever
+            filters_embedding=self.filters_embedding,
+            top_k_embedding=self.top_k_embedding,
+            filter_policy_embedding=self.filter_policy_embedding.value,
+            custom_query_embedding=self.custom_query_embedding,
+            # DocumentJoiner
+            join_mode=self.join_mode.value,
+            weights=self.weights,
+            top_k=self.top_k,
+            sort_by_score=self.sort_by_score,
+            # extra kwargs
+            **self.extra_args,
+        )
+
+    @classmethod
+    def from_dict(cls, data):
+        # deserialize the document store
+        doc_store = OpenSearchDocumentStore.from_dict(data["init_parameters"]["document_store"])
+        data["init_parameters"]["document_store"] = doc_store
+
+        # deserialize the embedder
+        try:
+            text_embedder_class = import_class_by_name(data["init_parameters"]["embedder"]["type"])
+        except ImportError as e:
+            msg = f"Class '{data['init_parameters']['embedder']['type']}' not correctly imported"
+            raise DeserializationError(msg) from e
+
+        data["init_parameters"]["embedder"] = component_from_dict(
+            cls=text_embedder_class, data=data["init_parameters"]["embedder"], name="embedder"
+        )
```

Comment:
It would be nice if we could add a `deserialize_text_embedder_inplace` like we have for `deserialize_chatgenerator_inplace` in haystack main. 

So probably not to further block this, let's open an issue to create `deserialize_text_embedder_inplace` in main and use that here in a later version.